### PR TITLE
Task/mpd 3648 adding services in presentation request service

### PR DESCRIPTION
**Pull request by gongte1937**
Add `getPresentationRequestResponseItem`,`getPresentationRequestResponseItems` and `deletePresentationRequestResponseItem` methods to `PresentationRequestResponseService`

### CODE REVIEW

File: packages/sdk/src/services/presentation-request-service.ts [hunk 1]
```diff
@@ -100,6 +100,41 @@ export class PresentationRequestService extends Service<PresentationRequestsApi>
     return itemService.create(itemServiceAuth, newPresentationRequestResponseItem);
   }
 
+  public async getPresentationRequestResponseItem(
+    auth: IVaultToken & IKeystoreToken & IDEK & IKEK,
+    responseId: string
```
Comment (vijayshiyani) [inline packages/sdk/src/services/presentation-request-service.ts]: please change from `responseId ` to `itemId` as it has nothing to do with the response its a vault item ID

File: packages/sdk/src/services/presentation-request-service.ts [hunk 2]
```diff
@@ -100,6 +100,41 @@ export class PresentationRequestService extends Service<PresentationRequestsApi>
...
+    auth: IVaultToken & IDEK,
+    credentialRequestId: string
+  ) {
+    const itemService = new ItemService(this.environment);
+    const itemServiceAuth = {
+      vault_access_token: auth.vault_access_token,
+      data_encryption_key: auth.data_encryption_key,
+    };
+    const listFilterOptions = {
+      name: credentialRequestId,
```
Comment (vijayshiyani) [inline packages/sdk/src/services/presentation-request-service.ts]: shouldn't this be `this.formatIdToItemName(presentationRequestId)`

File: packages/sdk/src/services/presentation-request-service.ts [hunk 3]
```diff
@@ -100,6 +100,41 @@ export class PresentationRequestService extends Service<PresentationRequestsApi>
...
+      data_encryption_key: auth.data_encryption_key,
+      key_encryption_key: auth.data_encryption_key,
+    };
+
+    return itemService.get(itemServiceAuth, responseId);
+  }
+
+  public async getPresentationRequestResponseItems(
+    auth: IVaultToken & IDEK,
+    credentialRequestId: string
```
Comment (vijayshiyani) [inline packages/sdk/src/services/presentation-request-service.ts]: please rename it to `presentationRequestId`

File: packages/sdk/src/services/presentation-request-service.ts [hunk 4]
```diff
@@ -100,6 +100,41 @@ export class PresentationRequestService extends Service<PresentationRequestsApi>
...
+      data_encryption_key: auth.data_encryption_key,
+    };
+    const listFilterOptions = {
+      name: credentialRequestId,
+    };
+
+    return itemService.listDecrypted(itemServiceAuth, listFilterOptions);
+  }
+
+  public async deletePresentationRequestResponseItem(auth: IVaultToken & IDEK, responseId: string) {
```
Comment (vijayshiyani) [inline packages/sdk/src/services/presentation-request-service.ts]: please change from `responseId`  to `itemId`

**Review by vijayshiyani — CHANGES_REQUESTED** [review]

**Review by vijayshiyani — APPROVED** [review]

**Comment by leikind** [comment]
@CodiumAI-Agent /improve

**Comment by leikind** [comment]
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 3 🔵🔵🔵⚪⚪</td></tr>
<tr><td>🧪&nbsp;<strong>PR contains tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<details><summary><a href='https://github.com/Meeco/js-sdk/pull/296/files#diff-1300c5cb1c462c7f56d17c1624889e071b9292db112ffbf1604fe97408286a64R112-R112'><strong>Possible Issue</strong></a>

In the getPresentationRequestResponseItem method, the key_encryption_key is incorrectly set to auth.data_encryption_key instead of auth.key_encryption_key. This could cause encryption/decryption issues.
</summary>

```typescript
key_encryption_key: auth.data_encryption_key,
```

</details>

<details><summary><a href='https://github.com/Meeco/js-sdk/pull/296/files#diff-50ab2c197046e9288d85426881cfd7cdbd126007f5d2ee47ec62aa5e3678c636R71-R79'><strong>Test Issue</strong></a>

The test for deletePresentationRequestResponseItem has a typo in the URL path - 'reponse_id' should be 'response_id'. This could indicate the actual implementation might have the same typo or the test is not properly validating the correct endpoint.
</summary>

```typescript
    .delete('/items/reponse_id')
    .matchHeader('Authorization', '2FPN4n5T68xy78i6HHuQ')
    .matchHeader('Meeco-Subscription-Key', 'environment_subscription_key')
    .reply(204)
)
.it('should delete the presentation response item by credentialRequestId', async () => {
  await new PresentationRequestService(environment).deletePresentationRequestResponseItem(
    testUserAuth,
    'reponse_id'
```

</details>

</td></tr>
</table>

**Comment by leikind** [comment]
## PR Code Suggestions ✨



<table><thead><tr><td><strong>Category</strong></td><td align=left><strong>Suggestion&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></td><td align=center><strong>Impact</strong></td></tr><tbody><tr><td rowspan=1>Possible issue</td>
<td>



<details><summary>Fix incorrect key assignment</summary>

___


**The <code>key_encryption_key</code> is incorrectly assigned the value of <code>data_encryption_key</code>. <br>This should use the <code>key_encryption_key</code> property from the auth parameter to maintain <br>proper encryption key separation.**

[packages/sdk/src/services/presentation-request-service.ts [112]](https://github.com/Meeco/js-sdk/pull/296/files#diff-1300c5cb1c462c7f56d17c1624889e071b9292db112ffbf1604fe97408286a64R112-R112)

```diff
-key_encryption_key: auth.data_encryption_key,
+key_encryption_key: auth.key_encryption_key,
```
<details><summary>Suggestion importance[1-10]: 9</summary>

__

Why: This is a critical bug where `key_encryption_key` is incorrectly assigned `auth.data_encryption_key` instead of `auth.key_encryption_key`. This could cause encryption/decryption failures and security issues.


</details></details></td><td align=center>High

</td></tr></tr></tbody></table>

**Pull request merged**